### PR TITLE
[genai_optimizations] Replace license-conflicting implementations

### PR DESCRIPTION
**Pull request by allnes**
## Summary

- Replace the copied MME metric calculator with `scikit-learn` metrics and local response grouping.
- Replace the copied math answer parser and grader with `math-verify` and use the official `openai/gsm8k` dataset.
- Remove the XAttention mode and antidiagonal block selection implementation; retain dense and Tri-Shape attention backed by the BSD-3-Clause Block-Sparse-Attention kernel.
- Pin Block-Sparse-Attention to commit `49d6c39e4dc0303442cda3bb758b3925d4399c49` and enforce its 128-token block size.
- Fall back to dense attention when a token-level padding mask or custom attention bias cannot be represented by the block-sparse kernel.
- Accept common leading MME answer prefixes while rejecting yes/no occurrences in arbitrary response text.
- Remove parser-only `latex2sympy2` and `word2number` dependencies and constrain Transformers to the 4.x API used by the module.
- Add third-party notices for the module dependencies and retained licensed sources.
- Checkout nested OpenVINO submodules recursively in Linux, macOS, and Windows workflows.

### CODE REVIEW

**Review by apaniukov — APPROVED** [review]

File: modules/genai_optimizations/genai_opt/sparse_attention.py [hunk 1]
```diff
@@ -1,774 +1,478 @@
...
+            q_offsets,
+            k_offsets,
+            head_mask_type,
+            None,
+            block_mask.contiguous(),
+            q_len,
+            k_len,
             p_dropout=0.0,
             deterministic=True,
+            softmax_scale=scaling,
```
[the remark this answers is not in the document]
Comment (allnes) [reply modules/genai_optimizations/genai_opt/sparse_attention.py:225]: No change needed. The pinned Block-Sparse-Attention implementation explicitly replaces softmax_scale=None with q.shape[-1] ** (-0.5) before invoking the CUDA kernel: https://github.com/mit-han-lab/Block-Sparse-Attention/blob/49d6c39e4dc0303442cda3bb758b3925d4399c49/block_sparse_attn/block_sparse_attn_interface.py#L352-L353. This matches the dense path.

File: modules/genai_optimizations/benchmarks/mme_metrics.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    recall_score,
+)
+
+VALID_ANSWERS = frozenset({"yes", "no"})
+
+
+def parse_yes_no(text: str) -> str:
+    """Return the first yes/no answer in a response, or ``other``."""
+    match = re.match(r"\s*[\W_]*(yes|no)\b", text, flags=re.IGNORECASE)
+    return match.group(1).lower() if match else "other"
```
[the remark this answers is not in the document]
Comment (allnes) [reply modules/genai_optimizations/benchmarks/mme_metrics.py:25]: Fixed in 8e31b7fe. The parser now accepts short leading prefixes such as Answer: yes, The answer is no, and Markdown-formatted **Answer:** yes, while still rejecting yes/no occurrences in arbitrary reasoning text.

File: modules/genai_optimizations/genai_opt/sparse_attention.py [hunk 3]
```diff
@@ -1,774 +1,478 @@
...
+        weights = F.softmax(weights, dim=-1, dtype=torch.float32).to(query_states.dtype)
+        output = torch.matmul(weights, value_states).transpose(1, 2).contiguous()
+        return output, weights if self.output_attentions else None
 
-    def _sparse_forward_common(
+    def _block_sparse_attention(
         self,
         module: nn.Module,
         query_states: torch.Tensor,
         key_states: torch.Tensor,
```
[the remark this answers is not in the document]
Comment (allnes) [reply modules/genai_optimizations/genai_opt/sparse_attention.py:155]: Fixed in 8e31b7fe. Tri-Shape now detects masking or bias on causally visible positions and falls back to dense attention. Pure causal masks remain on the block-sparse path. Tests cover both cases.

**Review by akashchi — APPROVED** [review]
GHA changes LGTM.
